securesystemslib/signer/_gcp_signer.py

```diff
@@ -163,24 +163,22 @@ def _get_hash_algorithm(public_key: Key) -> str:
         # TODO: This could be a public abstract method on Key so that GCPSigner
         # would not be tied to a specific Key implementation -- not all keys
         # have a pre hash algorithm though.
-        if public_key.keytype == "rsa":
-            # hash algorithm is encoded as last scheme portion
-            algo = public_key.scheme.split("-")[-1]
-        elif public_key.keytype in [
-            "ecdsa",
-            "ecdsa-sha2-nistp256",
-            "ecdsa-sha2-nistp384",
-        ]:
+        if (
+            public_key.keytype == "rsa" and public_key.scheme.endswith(("256", "512"))
+        ) or (
             # nistp256 uses sha-256, nistp384 uses sha-384
-            bits = public_key.scheme.split("-nistp")[-1]
-            algo = f"sha{bits}"
+            # TODO: Check for invalid type/scheme combinations (#766)
+            public_key.keytype
+            in ["ecdsa", "ecdsa-sha2-nistp256", "ecdsa-sha2-nistp384"]
+            and public_key.scheme.endswith(("256", "384"))
+        ):
+            algo = public_key.scheme[-3:]
```

Comment:
I'm a little surprised if this works since it seems to change the algo value ("sha256" -> "256"). 

The GCP tests currently only run on push so you can trigger it by pushing the branch to main repo

I'm fine with the simpler solution too.

Reply:
Good catch. 🤦  That happens when code isn't tested right away.